File: educational-core/src/com/jetbrains/edu/learning/courseFormat/tasks/EduTask.java
package com.jetbrains.edu.learning.courseFormat.tasks;

import com.jetbrains.edu.learning.courseFormat.CheckStatus;
import org.jetbrains.annotations.NotNull;

import java.util.Date;

/**
 * Original Edu plugin tasks with local tests and answer placeholders
 */
public class EduTask extends Task {

  public EduTask() {
  }

  public EduTask(@NotNull final String name, int id, int position, @NotNull Date updateDate, @NotNull CheckStatus status) {
    super(name, id, position, updateDate, status);
  }

  @Override
  public String getItemType() {
    return "edu";
  }

  @Override
  public boolean isToSubmitToStepik() {
    return myStatus != CheckStatus.Unchecked;
  }

  @Override
  public boolean supportSubmissions() {
    return true;
  }
}

File: educational-core/src/com/jetbrains/edu/learning/courseFormat/tasks/EduTask.kt
package com.jetbrains.edu.learning.courseFormat.tasks

import com.jetbrains.edu.learning.courseFormat.CheckStatus
import java.util.*

/**
 * Original Edu plugin tasks with local tests and answer placeholders
 */
open class EduTask : Task {
  constructor()
  constructor(name: String) : super(name)
  constructor(name: String, id: Int, position: Int, updateDate: Date, status: CheckStatus) :
    super(name, id, position, updateDate, status)

  override fun getItemType(): String = EDU_TASK_TYPE

  override fun isToSubmitToRemote(): Boolean {
    return myStatus != CheckStatus.Unchecked
  }

  override fun supportSubmissions(): Boolean = true

  companion object {
    @NonNls
    const val EDU_TASK_TYPE: String = "edu"

    @NonNls
    const val PYCHARM_TASK_TYPE: String = "pycharm"
  }
}